src/bdk_cli.rs

```diff
@@ -136,22 +206,46 @@ fn main() {
                             continue;
                         }
                         rl.add_history_entry(line.as_str());
-                        let split_line: Vec<&str> = line.split(' ').collect();
-                        let repl_subcommand: Result<ReplOpt, clap::Error> =
-                            ReplOpt::from_iter_safe(split_line);
+                        let split_line: Vec<&str> =
+                            split_regex.find_iter(&line).map(|m| m.as_str()).collect();
+                        let filtered_line: Vec<&str> = split_line
+                            .iter()
+                            .flat_map(|s| filter_regex.find_iter(s).map(|m| m.as_str()))
+                            .collect();
+                        let repl_subcommand: Result<ReplSubCommand, clap::Error> =
+                            ReplSubCommand::from_iter_safe(filtered_line);
                         debug!("repl_subcommand = {:?}", repl_subcommand);
 
                         if let Err(err) = repl_subcommand {
                             println!("{}", err.message);
                             continue;
                         }
 
-                        let result = bdk_cli::handle_wallet_subcommand(
-                            &Arc::clone(&wallet),
-                            repl_subcommand.unwrap().subcommand,
-                        )
-                        .unwrap();
-                        println!("{}", serde_json::to_string_pretty(&result).unwrap());
+                        let repl_subcommand = repl_subcommand.unwrap();
+
+                        let result = match repl_subcommand {
+                            ReplSubCommand::OnlineWalletSubCommand(online_subcommand) => {
+                                bdk_cli::handle_online_wallet_subcommand(
+                                    &Arc::clone(&online_wallet),
```

Comment:
I think you could just use `&online_wallet` here and in all the other places

Reply:
Actually, I think you could remove the `Arc` stuff entirely.. I don't think we are using them anymore, we are just passing normal references around

Reply:
I removed the Arc and it works fine.